pkg/boot/systembooter/bootentry.go

```diff
@@ -28,38 +29,42 @@ type BootEntry struct {
 	Booter Booter
 }
 
-var supportedBooterParsers = []func([]byte) (Booter, error){
+var supportedBooterParsers = []func([]byte, ulog.Logger) (Booter, error){
+	NewPxeBooter,
+	NewBootBooter,
 	NewNetBooter,
 	NewLocalBooter,
 }
 
 // GetBooterFor looks for a supported Booter implementation and returns it, if
 // found. If not found, a NullBooter is returned.
-func GetBooterFor(entry BootEntry) Booter {
+func GetBooterFor(entry BootEntry, l ulog.Logger) Booter {
 	var (
 		booter Booter
 		err    error
 	)
 	for idx, booterParser := range supportedBooterParsers {
 		log.Printf("Trying booter #%d", idx)
-		booter, err = booterParser(entry.Config)
+		booter, err = booterParser(entry.Config, l)
 		if err != nil {
 			log.Printf("This config is not valid for this booter (#%d)", idx)
+			log.Printf("  Error: %v", err.Error())
 			continue
 		}
 		break
 	}
 	if booter == nil {
-		log.Printf("No booter found for entry: %+v", entry)
+		log.Printf("No booter found for entry: %s: %s", entry.Name, string(entry.Config))
```

Comment:
HI, I am sorry I keep missing these log.Printf calls in packages, since I know this is slowing you down, but can you remove these log.Printf calls. Also, could this function return (Booter, error)  instead of NullBooter when nothing is found? That would be more idiomatic Go style.

Reply:
Done